### PR TITLE
Backlog 6419

### PR DESCRIPTION
**Pull request by scottyaslan**
I have no idea why this cant auto merge I started the day with a 'git reset --hard carlos/feature/BACKLOG-6419' so it should be fine....

### CODE REVIEW

File: package-res/resources/web/pentaho/lang/Filter.js [hunk 1]
```diff
@@ -1,42 +1,138 @@
...
+          filteredRows.push(k);
+        }
+      }
+
+      var dataView = new TableView(dataTable);
+      dataView.setSourceRows(filteredRows);
+      return dataView;
+    }
+  }, {
+    contains: function (filter, dataTable) {
```
Comment (carlosrusso) [inline package-res/resources/web/pentaho/lang/Filter.js:52]: contains is supposed to operate on a single row, not the whole list of rows.

File: test-js/unit/pentaho/lang/Filter.Spec.js [hunk 2]
```diff
@@ -14,61 +14,297 @@
...
+        });
+
+      });
+    });
+
+    describe("contains ", function() {
+
+      it("should return the contains of a simple filter", function() {
+        var productFilter = new Filter.IsIn("product", ["A", "B", "C", "D", "E", "F", "G"]);
+        var filteredData = productFilter.filter(data);
```
Comment (carlosrusso) [inline test-js/unit/pentaho/lang/Filter.Spec.js:257]: why bother adding the line `var filteredData = productFilter.filter(data);` if `filteredData` is not being used?

File: test-js/unit/pentaho/lang/Filter.Spec.js [hunk 3]
```diff
@@ -14,61 +14,297 @@
...
+
+      it("should return the contains of a simple filter", function() {
+        var productFilter = Filter.isIn("product", ["A", "B", "C", "D", "E", "F"]);
+        var filteredData = productFilter.filter(data);
+        expect(productFilter.contains(data)).toBe(false);
+      });
+
+    });
+
+    describe("AbstractTreeFilter ", function() {
```
Comment (carlosrusso) [inline test-js/unit/pentaho/lang/Filter.Spec.js]: these tests add no value

File: test-js/unit/pentaho/lang/Filter.Spec.js [hunk 4]
```diff
@@ -14,61 +14,297 @@
...
+          "B", "D", "E", "F", "G"
+        ].forEach(function(product, idx) {
+          expect(filteredData.getValue(idx, 0)).toBe(product);
+        });
+
+      });
+
+      it("should return the negation of a complex filter", function() {
+        var filter = new Filter.Or();
+        expect(filter.type).toBe("or");
```
Comment (carlosrusso) [inline test-js/unit/pentaho/lang/Filter.Spec.js:212]: tests on .type should stand on their own

File: package-res/resources/web/pentaho/lang/Filter.js (hunk 1 above)
Comment (scottyaslan) [inline package-res/resources/web/pentaho/lang/Filter.js:52]: sure, I was just using the dataTable as a placeholder while we were attempting to understand the contains....

File: test-js/unit/pentaho/lang/Filter.Spec.js (hunk 4 above)
Comment (scottyaslan) [inline test-js/unit/pentaho/lang/Filter.Spec.js:212]: agreed

File: test-js/unit/pentaho/lang/Filter.Spec.js (hunk 2 above)
Comment (scottyaslan) [inline test-js/unit/pentaho/lang/Filter.Spec.js:257]: I thought I was going to need this but it can be removed

**Pull request closed without merging**